apache/incubator-zeppelin
# Add checkbox as a type of dynamic forms

### What is this PR for?
1. Add checkbox support in dynamic forms
2. Fix ZEPPELIN-699: Cannot select the first item of dynamic forms when it is just created

### What type of PR is it?
Feature & Bug Fix

### Todos
* [x] Compatibility test with notes from previous versions
* [x] Nice CSS layout
* [x] Support checkbox creation by string substitution
* [x] Support pyspark

### Is there a relevant Jira issue?

https://issues.apache.org/jira/browse/ZEPPELIN-671
https://issues.apache.org/jira/browse/ZEPPELIN-669

### How should this be tested?
1. Create a %spark paragraph: `val a = z.checkbox("my_check", Seq(("a", "a"), ("b", "b"), ("c", "c")))`
2. Toggle the checkboxes and see the outcome

### Screenshots (if appropriate)
![igojnqh0mz](https://cloud.githubusercontent.com/assets/3282033/13136828/929ec1ec-d5d1-11e5-89d9-98659f444f37.gif)

### Questions:
* Does the licenses files need update?

NO

* Is there breaking changes for older versions?

Sort of. I am not sure whether Input.type (form.type) has already been used in other places

* Does this needs documentation?

YES
